pallets/ajuna-awesome-avatars/src/types.rs

```diff
@@ -44,10 +46,37 @@ pub struct Season<BlockNumber> {
 	pub max_mythical_mints: u16,
 	pub rarity_tiers: RarityTiers,
 	pub max_variations: u8,
+	pub max_components: u32,
 }
 
 #[derive(Encode, Decode, MaxEncodedLen, TypeInfo, Clone, Debug, Default, Eq, PartialEq)]
 pub struct SeasonMetadata {
 	pub name: BoundedVec<u8, ConstU32<100>>,
 	pub description: BoundedVec<u8, ConstU32<1000>>,
 }
+
+pub const DNA_VERSION: u8 = 0;
+pub const MAX_COMPONENTS: u32 = 100;
+
+#[derive(Encode, Decode, Default, Clone, MaxEncodedLen, PartialEq, TypeInfo)]
+pub struct Dna(BoundedVec<u8, ConstU32<MAX_COMPONENTS>>);
+
+impl TryFrom<Vec<u8>> for Dna {
+	type Error = ();
+
+	fn try_from(value: Vec<u8>) -> Result<Self, Self::Error> {
+		if let Ok(dna) = BoundedVec::try_from(value) {
+			Ok(Self(dna))
+		} else {
+			Err(())
+		}
+	}
+}
```

Comment:
I think just having a type alias should work because `BoundedVec` should have `try_from` method implemented
```suggestion
type Dna = BoundedVec<u8, ConstU32<MAX_COMPONENTS>>;
```